pil-analyzer/tests/types.rs

```diff
@@ -734,3 +734,39 @@ fn prover_functions() {
     ";
     type_check(input, &[("a", "", "int"), ("b", "", "()[]")]);
 }
+
```

Comment:
Please add a test like `enum A { X } let a = A{x: 8};`, i.e. where you use an enum for a struct expression.